modules/server/src/main/java/org/jboss/wsf/stack/cxf/security/authentication/SubjectCreator.java

```diff
@@ -86,38 +95,46 @@ public Subject createSubject(SecurityDomainContext ctx, String name, String pass
       }
 
       // authenticate and populate Subject
-      
-
       Principal principal = new SimplePrincipal(name);
       Subject subject = new Subject();
 
       boolean TRACE = SECURITY_LOGGER.isTraceEnabled();
       if (TRACE)
          SECURITY_LOGGER.aboutToAuthenticate(ctx.getSecurityDomain());
 
-      try
-      {
-         ClassLoader tccl = SecurityActions.getContextClassLoader();
-         //allow PicketBox to see jbossws modules' classes
-         SecurityActions.setContextClassLoader(createDelegateClassLoader(ClassLoaderProvider.getDefaultProvider().getServerIntegrationClassLoader(), tccl));
-         try
-         {
-            if (ctx.isValid(principal, password, subject) == false)
-            {
+      try {
+         SecurityDomain securityDomain = ctx.getSecurityDomainObject();
```

Comment:
Should this be  ctx.getSecurityDomainObject()?

Reply:
@jimma Fixed to be `getElytronSecurityDomain`